modules/ml/src/main/java/org/apache/ignite/ml/inference/ModelSignature.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.inference;
+
+import java.io.Serializable;
+
+/**
+ * Signature that defines input/output types in Protobuf.
+ */
+public class ModelSignature implements Serializable {
```

Comment:
The key point here is to define a skeleton for inference/serve that handles TensorFlow case. I think that it makes sense to keep this part in this PR so that the whole inference infrastructure is defined without contradictions.

Reply:
Great to separate on two PRs, agree with @avplatonov